functions/v2/typed/googlechatapp/main.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_typed_googlechatbot]
+import functions_framework
+
+
+class ChatRequest:
+    message: dict
+
+    # Required to deserialize the request
+    @staticmethod
+    def from_dict(req: dict) -> "ChatRequest":
+        obj = ChatRequest()
+        obj.message = req["message"]
+        return obj
+
+
+class ChatResponse:
+    cardId: str
+    card: dict
```

Comment:
could you clarify more than just a `dict`? What type of `dict` is it? Try to be concise and be liberal when using `Any`.

Reply:
I think I can decompose this into more classes or we could possibly use `TypedDict` in more recent versions of python (not sure if this will work with 3.7). Do you have a recommendation here?

Reply:
Solely based on the example shown in this code it looks like the type might be
```
dict[str, Union[str, dict[str, str], list[dict[str, str]])
```
This already looks complicated however, and if you expect more than just the Union of the three (str, dict, list[dict]) then `dict[str, Any]` would be more appropriate. If it's just the union of the three, maybe we can mix it up with a type alias, something like

```
from typing import Dict, List, TypeAlias, Union
_messageItemType: TypeAlias = Dict[str, str]
...

class ChatRequest:
    message: Dict[str, Union[str, _messageItemType, list[_messageItemType]]]
...
```

Reply:
Thanks for illustrating those! I went ahead and adopted
```
class ChatRequest:
    message: Dict[str, Union[str, _messageItemType, List[_messageItemType]]]
```
as the type annotation.